File: app/src/main/java/com/hover/stax/onboarding/SlidesPagerAdapter.java
package com.hover.stax.onboarding;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.hover.stax.R;

import org.jetbrains.annotations.NotNull;

public class SlidesPagerAdapter extends FragmentPagerAdapter {
    private static final int SLIDES_SIZE = 3;

    public SlidesPagerAdapter(@NonNull FragmentManager fm, int behavior) {
        super(fm, behavior);
    }

    @Override
    public int getCount() {
        return SLIDES_SIZE;
    }

    @NotNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return SlidesFragment.newInstance(R.layout.slide1_fragment);
            case 1:
                return SlidesFragment.newInstance(R.layout.slide2_fragment);
            case 2:
                return SlidesFragment.newInstance(R.layout.slide3_fragment);
            default:
                return new Fragment();
        }
    }
}

File: app/src/main/java/com/hover/stax/onboarding/SlidesPagerAdapter.kt
package com.hover.stax.onboarding

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.hover.stax.R

class SlidesPagerAdapter(fragmentManager: FragmentManager, behaviour: Int) : FragmentPagerAdapter(fragmentManager, behaviour) {

    override fun getCount(): Int = SLIDES_SIZE

    override fun getItem(position: Int): Fragment = when (position) {
        0 -> SlidesFragment.newInstance(R.layout.slide1_fragment)
        1 -> SlidesFragment.newInstance(R.layout.slide2_fragment)
        2 -> SlidesFragment.newInstance(R.layout.slide3_fragment)
        else -> Fragment()
    }

    companion object {
        const val SLIDES_SIZE = 3
    }
}